netket/abstract_variational_driver.py

```diff
@@ -106,6 +106,23 @@ def machine(self):
         """
         return self._machine
 
+    @property
+    def step_count(self):
+        """
+        Returns a monotonic integer labelling all the steps performed by this driver.
+        This can be used, for example, to identify the line in a log file.
+        """
+        return self._step_count
+
+    @property
+    def step_value(self):
```

Comment:
why do we need both step_count and step_value?

Reply:
If we want to support with the same `AbstractVariationalDriver` a time-integrator, we need to have both.
`Step_value` is like time and `step_count` is really the integer step.

This would allow to remove a lot of duplicate code from `exact_time_evo`, btw.

Reply:
Maybe instead of the step value there should be an abstract method htat returns a dict of auxiliary information to be displayed. (Maybe something like `driver.auxiliary_info()`). By default this can be empty (or maybe we can put the current energy stats there for VMC, if that makes the code simpler), for the exact time evolution it could additionally include `{"t": current_time}`, etc. This can then also be used in by `set_postfix` the progress bar.

Reply:
What you are referring to is achieved by `_log_additional_data`, which appends to the log_dict more stuff to log.

Step_value is really for things like the progress bar and for the integrator. Did we finish yet or not? How many steps should we take? If the integrator has fixed steps, we can always use a 1 to 1 correspondence between step_count and time, but if it's not fixed, or if you change it across iterations manually, this becomes harder to achieve.

Reply:
Okay, I can see the point of singling out the `step_value` since it is used in `progress_bar.update`; that does make sense.